crawl-ref/source/mon-data.h

```diff
@@ -5909,7 +5909,7 @@ DUMMY(MONS_GOLEM, '8', LIGHTGREY, "golem", TILEP_MONS_IRON_GOLEM)
 
 { // always redefined
     MONS_STATUE, '8', LIGHTGREY, "statue",
-    M_STATIONARY | M_SPEAKS | M_ARCHER | M_DONT_MELEE | M_NO_POLY_TO,
+    M_STATIONARY | M_SPEAKS | M_ARCHER | M_DONT_MELEE | M_NO_POLY_TO | M_NO_REGEN,
```

Comment:
`mons_can_regenerate()` checks the monster class's flags as a fallback, so you should be able to remove all other `M_NO_REGEN` flags and just keep the one on `MONS_STATUE`. Haven't tested this though.

Reply:
I tested this and it does not appear to work - Only `MONS_STATUE` and its derivatives keep the "It cannot regenerate" text, all other statues do not get it.